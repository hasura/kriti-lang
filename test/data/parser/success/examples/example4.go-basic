[{{ range i, x := $.events }} { "name": {{ x.name }} } {{ end }}]