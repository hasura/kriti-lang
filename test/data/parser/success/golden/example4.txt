Array
    [ Range
        ( SourcePosition
            { _sourceName = "sourceName"
            , _line = 1
            , _column = 2
            }
        , Just
            ( SourcePosition
                { _sourceName = "sourceName"
                , _line = 1
                , _column = 65
                }
            )
        )
        ( Just "i" ) "x"
        [
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 1
                    , _column = 19
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 1
                        , _column = 20
                        }
                    )
                )
            , Obj "$"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 1
                    , _column = 20
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 1
                        , _column = 26
                        }
                    )
                )
            , Obj "events"
            )
        ]
        ( Object
            ( fromList
                [
                    ( "name"
                    , Path
                        [
                            (
                                ( SourcePosition
                                    { _sourceName = "sourceName"
                                    , _line = 1
                                    , _column = 44
                                    }
                                , Just
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 1
                                        , _column = 45
                                        }
                                    )
                                )
                            , Obj "x"
                            )
                        ,
                            (
                                ( SourcePosition
                                    { _sourceName = "sourceName"
                                    , _line = 1
                                    , _column = 45
                                    }
                                , Just
                                    ( SourcePosition
                                        { _sourceName = "sourceName"
                                        , _line = 1
                                        , _column = 49
                                        }
                                    )
                                )
                            , Obj "name"
                            )
                        ]
                    )
                ]
            )
        )
    ]